aws/resource_aws_ecs_service_test.go

```diff
@@ -618,13 +638,47 @@ func testAccCheckAWSEcsServiceDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccCheckAWSEcsServiceExists(name string) resource.TestCheckFunc {
+func testAccCheckAWSEcsServiceExists(name string, service *ecs.Service) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		_, ok := s.RootModule().Resources[name]
+		rs, ok := s.RootModule().Resources[name]
 		if !ok {
 			return fmt.Errorf("Not found: %s", name)
 		}
 
+		conn := testAccProvider.Meta().(*AWSClient).ecsconn
+
+		input := &ecs.DescribeServicesInput{
+			Cluster:  aws.String(rs.Primary.Attributes["cluster"]),
+			Services: []*string{aws.String(rs.Primary.ID)},
+		}
+		var output *ecs.DescribeServicesOutput
+		err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+			var err error
+			output, err = conn.DescribeServices(input)
```

Comment:
Throttling errors are another retryable possibility here.

Reply:
Throttling errors hopefully are retried already in the upstream SDK and if not, they should be 😉 

https://github.com/aws/aws-sdk-go/blob/master/aws/request/retryer.go#L35-L43